actors/ethaccount/src/lib.rs

```diff
@@ -0,0 +1,150 @@
+pub mod types;
+
+use fvm_ipld_encoding::ipld_block::IpldBlock;
+use fvm_shared::address::{Payload, Protocol};
+use fvm_shared::crypto::hash::SupportedHashes::Keccak256;
+use fvm_shared::error::ExitCode;
+use fvm_shared::{MethodNum, METHOD_CONSTRUCTOR};
+use num_derive::FromPrimitive;
+
+use crate::types::AuthenticateMessageParams;
+use fil_actors_runtime::runtime::{ActorCode, Runtime};
+use fil_actors_runtime::{
+    actor_dispatch, actor_error, ActorDowncast, ActorError, AsActorError, EAM_ACTOR_ID,
+    FIRST_EXPORTED_METHOD_NUMBER, SYSTEM_ACTOR_ADDR,
+};
+
+#[cfg(feature = "fil-actor")]
+fil_actors_runtime::wasm_trampoline!(EthAccountActor);
+
+/// Ethereum Account actor methods.
+#[derive(FromPrimitive)]
+#[repr(u64)]
+pub enum Method {
+    Constructor = METHOD_CONSTRUCTOR,
+    AuthenticateMessageExported = frc42_dispatch::method_hash!("AuthenticateMessage"),
+}
+
+/// Ethereum Account actor.
+pub struct EthAccountActor;
+
+impl EthAccountActor {
+    /// Ethereum Account actor constructor.
+    /// NOTE: This method is NOT currently called from anywhere, instead the FVM just deploys EthAccounts.
```

Comment:
Is this comment still accurate? I thought the EAM might be involved. I guess this (explicit Eth support/privilege in the VM itself) is something I missed in the FIPs.

Reply:
Yes, still accurate -- The EAM creates Placeholders and EVM actors, and the FVM just turns Placeholders into EthAccounts.

Reply:
Ok, there's some magic going on there where the FVM must know something special about EVM actors, but 🤞 we can de-privilege it later.